server/server.go

```diff
@@ -75,6 +83,42 @@ type Server struct {
 	defaultIDMappings *idtools.IDMappings
 }
 
+type certConfigCache struct {
+	config  *tls.Config
+	expires time.Time
+
+	tlsCert string
+	tlsKey  string
+	tlsCA   string
+}
+
+// GetConfigForClient gets the tlsConfig for the streaming server.
+// This allows the certs to be swapped, without shutting down crio.
+func (cc *certConfigCache) GetConfigForClient(hello *tls.ClientHelloInfo) (*tls.Config, error) {
```

Comment:
I don't think there is a great story for certificate rotation in Kubernetes right now. We are doing this, because we come from a hyper secure posture of dealing with healthcare compliance and we have an aggressively attenuated schedule for expiring our PKI leaf certificates. Given that is the case, it seems overly inconvenient to have to stop CRIO just to reload a certificate.

Netflix actually expires their server certificates [every 4 days](https://medium.facilelogin.com/short-lived-certificates-netflix-fd5f3ae5bc9) (or at least has proposed it as a security feature).